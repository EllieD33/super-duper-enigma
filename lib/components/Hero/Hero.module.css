.textContainer {
    max-width: 800px;
    min-width: 500px;
    height: 100%;
    display: flex;
    flex-direction: column;
    gap: 18px;
    justify-content: center;
}

.particlesContainer {
    max-height: 470px;
}

.heroGrid {
    display: grid;
    height: 100vh;
    background-color: #1b1b1b;
    background-image: linear-gradient(
        to left bottom,
        #1b1b1b,
        #231e2b,
        #2f1f38,
        #401d43,
        #541649
    );
}
.heroTopRow {
    display: flex;
    justify-content: space-between;
    width: 100%;
    gap: 40px;
    padding: 0 80px;
}

@media (max-width: 998px) {
    .heroTopRow {
        display: grid;
        place-items: center;
        grid-template-areas: "stack";
        padding: 0 20px;
    }

    .textContainer,
    .particlesContainer {
        grid-area: stack;
    }
    .particlesContainer {
        opacity: 0.6;
        z-index: -1;
    }
}
