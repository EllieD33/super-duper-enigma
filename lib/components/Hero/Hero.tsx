import React, { ReactElement } from "react";
import styles from "./Hero.module.css";
import Text from "../Text/Text";
import GradientHeader from "../GradientHeader/GradientHeader";
import { Colours } from "../../constants/Colours";
import { FontSize, FontWeight } from "../../constants/Typography";
import { TextTag } from "../../constants/TextTags";

export interface HeroProps {}

const Hero = ({}: HeroProps): ReactElement => {
    return (
        <>
            <Text
                size={FontSize.XXL}
                weight={FontWeight.ExtraBold}
                colour={Colours.White}
            >
                Experts in Web3{" "}
                <GradientHeader
                    as={TextTag.Span}
                    size={FontSize.XXL}
                    weight={FontWeight.ExtraBold}
                >
                    Security & Crisis Response
                </GradientHeader>
                .
            </Text>
            <Text
                styleAs="Heading2"
                weight={FontWeight.Medium}
                colour={Colours.White}
            >
                We're here when you need us.
            </Text>
            <Text styleAs="Heading2" colour={Colours.White}>
                Safer together.
            </Text>
        </>
    );
};

export default Hero;
