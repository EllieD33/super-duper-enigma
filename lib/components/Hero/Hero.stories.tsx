import type { Meta, StoryObj } from "@storybook/react";
import Hero, { HeroProps } from "./Hero";

const meta = {
    title: "Library/Hero/Hero",
    component: Hero,
    parameters: {
        layout: "fullscreen",
    },
    tags: ["autodocs"],
    argTypes: {},
} satisfies Meta<typeof Hero>;

export default meta;
type Story = StoryObj<typeof meta>;

const defaultProps = {};

export const Default: Story = {
    args: {
        ...defaultProps,
    },
};
