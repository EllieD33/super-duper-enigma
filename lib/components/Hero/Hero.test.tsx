import React from "react";
import { render, screen } from "@testing-library/react";
import Hero, { HeroProps } from "./Hero";

const defaultProps: HeroProps = {};

describe("Hero", () => {
    it("should render the default component", () => {
        const container = render(<Hero {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
});
