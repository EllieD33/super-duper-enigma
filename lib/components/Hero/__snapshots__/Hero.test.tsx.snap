// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render the default component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="left"
        style="color: rgb(249, 249, 249); font-weight: 800; font-size: 2rem; font-family: 'Open Sans', sans-serif;"
      >
        Experts in Web3
         
        <span
          class="left pinkToBlue"
          style="color: rgb(51, 51, 51); font-weight: 800; font-size: 2rem; font-family: 'Open Sans', sans-serif;"
        >
          Security & Crisis Response
        </span>
        .
      </p>
      <p
        class="left"
        style="color: rgb(249, 249, 249); font-weight: 500; font-size: 1.75rem; font-family: 'Roboto', sans-serif;"
      >
        We're here when you need us.
      </p>
      <p
        class="left"
        style="color: rgb(249, 249, 249); font-weight: 700; font-size: 1.75rem; font-family: 'Roboto', sans-serif;"
      >
        Safer together.
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="left"
      style="color: rgb(249, 249, 249); font-weight: 800; font-size: 2rem; font-family: 'Open Sans', sans-serif;"
    >
      Experts in Web3
       
      <span
        class="left pinkToBlue"
        style="color: rgb(51, 51, 51); font-weight: 800; font-size: 2rem; font-family: 'Open Sans', sans-serif;"
      >
        Security & Crisis Response
      </span>
      .
    </p>
    <p
      class="left"
      style="color: rgb(249, 249, 249); font-weight: 500; font-size: 1.75rem; font-family: 'Roboto', sans-serif;"
    >
      We're here when you need us.
    </p>
    <p
      class="left"
      style="color: rgb(249, 249, 249); font-weight: 700; font-size: 1.75rem; font-family: 'Roboto', sans-serif;"
    >
      Safer together.
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
